robot/impl/local_robot.go

```diff
@@ -402,19 +402,29 @@ func newWithResources(
 	}()
 
 	if cfg.Cloud != nil && cfg.Cloud.AppAddress != "" {
-		_, cloudConn, err := r.cloudConnSvc.AcquireConnection(ctx)
-		if err == nil {
-			r.packageManager, err = packages.NewCloudManager(cfg.Cloud, pb.NewPackageServiceClient(cloudConn), cfg.PackagePath, logger)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			if !errors.Is(err, context.DeadlineExceeded) {
-				return nil, err
+		connectionChan := make(chan packages.DeferredConnectionResponse)
+		goutils.PanicCapturingGo(func() {
+			for {
+				// deferred connection manager uses this for-loop behavior to
+				// implement connection retries
+				_, cloudConn, err := r.cloudConnSvc.AcquireConnection(ctx)
+				select {
+				case connectionChan <- packages.DeferredConnectionResponse{
+					Client: packagespb.NewPackageServiceClient(cloudConn),
+					Err:    err,
+				}:
+				case <-closeCtx.Done():
+					return
+				}
+				select {
+				// wait minimum 1 second before retrying to create a new connection
+				case <-time.After(time.Second * 1):
```

Comment:
this isn't a merge blocker, but I'm thinking about the connection timeout in the AcquireConnection call -- the retry frequency may be different depending on what kind of failure is happening

(i.e. disconnected network vs routing timeout)